diff --git a/node_modules/@graphql-codegen/gql-tag-operations/cjs/index.js b/node_modules/@graphql-codegen/gql-tag-operations/cjs/index.js
index 51a9272..032e0bf 100644
--- a/node_modules/@graphql-codegen/gql-tag-operations/cjs/index.js
+++ b/node_modules/@graphql-codegen/gql-tag-operations/cjs/index.js
@@ -91,9 +91,8 @@ function getDocumentRegistryChunk(sourcesWithOperations = []) {
     lines.add(` * Therefore it is highly recommended to use the babel or swc plugin for production.\n */\n`);
     lines.add(`const documents = {\n`);
     for (const { operations, ...rest } of sourcesWithOperations) {
-        const originalString = rest.source.rawSDL;
         const operation = operations[0];
-        lines.add(`    ${JSON.stringify(originalString)}: types.${operation.initialName},\n`);
+        lines.add(`    ${operation.initialName}: types.${operation.initialName},\n`);
     }
     lines.add(`};\n`);
     return lines;
@@ -102,8 +101,8 @@ function getGqlOverloadChunk(sourcesWithOperations, gqlTagName, mode, emitLegacy
     const lines = new Set();
     // We intentionally don't use a <T extends keyof typeof documents> generic, because TS
     // would print very long `gql` function signatures (duplicating the source).
-    for (const { operations, ...rest } of sourcesWithOperations) {
-        const originalString = rest.source.rawSDL;
+    for (const { operations } of sourcesWithOperations) {
+        const originalString = operations[0]?.initialName;
         const returnType = mode === 'lookup'
             ? `(typeof documents)[${JSON.stringify(originalString)}]`
             : emitLegacyCommonJSImports
