diff --git a/node_modules/@graphql-codegen/client-preset/cjs/index.js b/node_modules/@graphql-codegen/client-preset/cjs/index.js
index be26a67..dfc776e 100644
--- a/node_modules/@graphql-codegen/client-preset/cjs/index.js
+++ b/node_modules/@graphql-codegen/client-preset/cjs/index.js
@@ -116,6 +116,7 @@ export const preset = {
         ];
         const gqlArtifactFileExtension = '.ts';
         reexports.push('gql');
+        reexports.push('graphql');
         const config = {
             ...options.config,
             inlineFragmentTypes: isMaskingFragments ? 'mask' : options.config['inlineFragmentTypes'],
