diff --git a/node_modules/@intercom/intercom-react-native/android/src/main/java/com/intercom/reactnative/IntercomModule.java b/node_modules/@intercom/intercom-react-native/android/src/main/java/com/intercom/reactnative/IntercomModule.java
index f779387..d203e45 100644
--- a/node_modules/@intercom/intercom-react-native/android/src/main/java/com/intercom/reactnative/IntercomModule.java
+++ b/node_modules/@intercom/intercom-react-native/android/src/main/java/com/intercom/reactnative/IntercomModule.java
@@ -36,11 +36,12 @@ import io.intercom.android.sdk.push.IntercomPushClient;
 @ReactModule(name = IntercomModule.NAME)
 public class IntercomModule extends ReactContextBaseJavaModule {
   public static final String NAME = "IntercomModule";
-
+  private final ReactApplicationContext reactContext;
   private static final IntercomPushClient intercomPushClient = new IntercomPushClient();
 
   public IntercomModule(ReactApplicationContext reactContext) {
     super(reactContext);
+    this.reactContext = reactContext;
   }
 
   @Override
@@ -76,6 +77,15 @@ public class IntercomModule extends ReactContextBaseJavaModule {
     Log.d(NAME, "sendTokenToIntercom");
   }
 
+  @ReactMethod
+  public void init(String apiKey, String appId, Promise promise) {
+      String sdkVersion = BuildConfig.INTERCOM_VERSION_NAME;
+      ReactNativeHeaderInterceptor.setReactNativeVersion(this.reactContext.getApplicationContext(), sdkVersion);
+      Intercom.initialize(this.getCurrentActivity().getApplication(), apiKey, appId);
+      Log.d(NAME, "init");
+      promise.resolve(true);
+  }
+
   @ReactMethod
   public void handlePushMessage(Promise promise) {
     try {
diff --git a/node_modules/@intercom/intercom-react-native/ios/IntercomModule.m b/node_modules/@intercom/intercom-react-native/ios/IntercomModule.m
index 4600e41..360fa2f 100644
--- a/node_modules/@intercom/intercom-react-native/ios/IntercomModule.m
+++ b/node_modules/@intercom/intercom-react-native/ios/IntercomModule.m
@@ -72,6 +72,15 @@ - (NSData *)dataFromHexString:(NSString *)string {
     return commandToSend;
 }
 
+RCT_EXPORT_METHOD(init:
+                  (nonnull NSString *) apiKey:
+                  (nonnull NSString *) appId:
+                  (RCTPromiseResolveBlock) resolve :(RCTPromiseRejectBlock)reject) {
+    [IntercomModule initialize:apiKey withAppId:appId];
+    NSLog(@"initialized Intercom module: %@ - %@", apiKey, appId);
+    resolve(@(YES));
+};
+
 RCT_EXPORT_METHOD(sendTokenToIntercom:
                   (NSString *) token:
                   (RCTPromiseResolveBlock) resolve :(RCTPromiseRejectBlock)reject) {
diff --git a/node_modules/@intercom/intercom-react-native/lib/typescript/index.d.ts b/node_modules/@intercom/intercom-react-native/lib/typescript/index.d.ts
index f13d9d6..1d38ae7 100644
--- a/node_modules/@intercom/intercom-react-native/lib/typescript/index.d.ts
+++ b/node_modules/@intercom/intercom-react-native/lib/typescript/index.d.ts
@@ -91,6 +91,7 @@ export declare type IntercomType = {
     hideIntercom(): Promise<boolean>;
     logEvent(eventName: string, metaData?: MetaData): Promise<boolean>;
     logout(): Promise<boolean>;
+    init(apiKey: string, appId: string): Promise<boolean>;
     registerIdentifiedUser(params: Registration): Promise<boolean>;
     registerUnidentifiedUser(): Promise<boolean>;
     setBottomPadding(bottomPadding: number): Promise<boolean>;
diff --git a/node_modules/@intercom/intercom-react-native/src/index.tsx b/node_modules/@intercom/intercom-react-native/src/index.tsx
index 537f48d..37f7b24 100644
--- a/node_modules/@intercom/intercom-react-native/src/index.tsx
+++ b/node_modules/@intercom/intercom-react-native/src/index.tsx
@@ -114,6 +114,7 @@ export type IntercomType = {
   hideIntercom(): Promise<boolean>;
   logEvent(eventName: string, metaData?: MetaData): Promise<boolean>;
   logout(): Promise<boolean>;
+  init(apiKey: string, appId: string): Promise<boolean>;
   registerIdentifiedUser(params: Registration): Promise<boolean>;
   registerUnidentifiedUser(): Promise<boolean>;
   setBottomPadding(bottomPadding: number): Promise<boolean>;
@@ -154,6 +155,7 @@ const Intercom = {
   logEvent: (eventName, metaData = undefined) =>
     IntercomModule.logEvent(eventName, metaData),
   logout: () => IntercomModule.logout(),
+  init: (apiKey: string, appId: string) => IntercomModule.init(apiKey, appId),
   registerIdentifiedUser: (eventName) =>
     IntercomModule.registerIdentifiedUser(eventName),
   registerUnidentifiedUser: () => IntercomModule.registerUnidentifiedUser(),
