diff --git a/node_modules/@react-native-camera-roll/camera-roll/ios/RNCCameraRoll.mm b/node_modules/@react-native-camera-roll/camera-roll/ios/RNCCameraRoll.mm
index cd55c96..6cfc725 100644
--- a/node_modules/@react-native-camera-roll/camera-roll/ios/RNCCameraRoll.mm
+++ b/node_modules/@react-native-camera-roll/camera-roll/ios/RNCCameraRoll.mm
@@ -22,11 +22,6 @@
 
 #import "RNCAssetsLibraryRequestHandler.h"
 
-#if __has_include(<SDWebImageWebPCoder/SDWebImageWebPCoder.h>)
-  #import <SDWebImageWebPCoder/SDWebImageWebPCoder.h>
-  #define SD_WEB_IMAGE_WEBP_CODER_AVAILABLE 1
-#endif
-
 @implementation RCTConvert (PHAssetCollectionSubtype)
 
 RCT_ENUM_CONVERTER(PHAssetCollectionSubtype, (@{
@@ -175,21 +170,6 @@ static void requestPhotoLibraryAccess(RCTPromiseRejectBlock reject, PhotosAuthor
         assetRequest = request;
       } else {
         NSData *data = [NSData dataWithContentsOfURL:inputURI];
-        if ([[inputURI.pathExtension lowercaseString] isEqualToString:@"webp"]) {
-          UIImage *webpImage;
-
-          #ifdef SD_WEB_IMAGE_WEBP_CODER_AVAILABLE 
-            webpImage = [[SDImageWebPCoder sharedCoder] decodedImageWithData:data options:nil];
-          #else
-            if (@available(iOS 14, *)) {
-              webpImage = [UIImage imageWithData:data];
-            }
-          #endif
-          
-          if (webpImage) {
-            data = UIImageJPEGRepresentation(webpImage, 1.0);
-          }
-        }
         UIImage *image = [UIImage imageWithData:data];
         assetRequest = [PHAssetChangeRequest creationRequestForAssetFromImage:image];
       }
