diff --git a/node_modules/@wootric/react-native-wootric/index.d.ts b/node_modules/@wootric/react-native-wootric/index.d.ts
new file mode 100644
index 0000000..9892360
--- /dev/null
+++ b/node_modules/@wootric/react-native-wootric/index.d.ts
@@ -0,0 +1,16 @@
+class Wootric {
+    public static configureWithClientID: (clientId: string, secret: string) => void;
+    public static setEndUserEmail: (email: string) => void;
+    public static setSurveyImmediately: (isImmediately: boolean) => void;
+    public static setEndUserCreatedAt: (createdAt: number) => void;
+    public static setEndUserExternalId: (externalId: string) => void;
+    public static setEndUserPhoneNumber: (phoneNr: string) => void;
+    public static setEndUserProperties: (props: any) => void;
+    public static showOptOut: (optOut: boolean) => void;
+    public static setFirstSurveyAfter: (someNumber: number) => void;
+    public static setCustomLanguage: (lang: string) => void;
+    public static setCustomProductName: (productName: string) => void;
+    public static setCustomAudience: (name: string) => void;
+    public static showSurvey: () => void;
+}
+export default Wootric;
\ No newline at end of file
diff --git a/node_modules/@wootric/react-native-wootric/ios/RNWootric.podspec b/node_modules/@wootric/react-native-wootric/ios/RNWootric.podspec
index 5ff3503..6f5c17b 100644
--- a/node_modules/@wootric/react-native-wootric/ios/RNWootric.podspec
+++ b/node_modules/@wootric/react-native-wootric/ios/RNWootric.podspec
@@ -12,8 +12,10 @@ Pod::Spec.new do |s|
   s.author             = { "author" => "author@domain.cn" }
   s.platform     = :ios, "7.0"
   s.source       = { :git => "https://github.com/author/RNWootric.git", :tag => "master" }
-  s.source_files  = "RNWootric/**/*.{h,m}"
+  s.source_files  = "*.{h,m}"
+  s.public_header_files  = "*.{h,m}"
   s.requires_arc = true
+  s.homepage  = "http://google.com"
 
 
   s.dependency "React"
diff --git a/node_modules/@wootric/react-native-wootric/package.json b/node_modules/@wootric/react-native-wootric/package.json
index 5d9c18f..c5c09f8 100644
--- a/node_modules/@wootric/react-native-wootric/package.json
+++ b/node_modules/@wootric/react-native-wootric/package.json
@@ -16,5 +16,6 @@
   },
   "homepage": "https://github.com/Wootric/react-native-wootric#readme",
   "author": "Wootric",
-  "license": "MIT"
+  "license": "MIT",
+  "types": "./index.d.ts"
 }
