diff --git a/node_modules/@yu-life/react-native-yu-health/android/src/main/java/com/yuhealth/YuHealthModule.kt b/node_modules/@yu-life/react-native-yu-health/android/src/main/java/com/yuhealth/YuHealthModule.kt
index 2fd96ab..c7aaa8c 100644
--- a/node_modules/@yu-life/react-native-yu-health/android/src/main/java/com/yuhealth/YuHealthModule.kt
+++ b/node_modules/@yu-life/react-native-yu-health/android/src/main/java/com/yuhealth/YuHealthModule.kt
@@ -45,7 +45,7 @@ class YuHealthModule internal constructor(private var context: ReactApplicationC
   }
 
   @ReactMethod
-  override fun removeListeners(count: Int) {
+  override fun removeListeners(count: Double) {
     // Required definition for TurboModules
   }
 
diff --git a/node_modules/@yu-life/react-native-yu-health/ios/YuHealth.h b/node_modules/@yu-life/react-native-yu-health/ios/YuHealth.h
index befd565..5863cd4 100644
--- a/node_modules/@yu-life/react-native-yu-health/ios/YuHealth.h
+++ b/node_modules/@yu-life/react-native-yu-health/ios/YuHealth.h
@@ -2,7 +2,7 @@
 #import <CoreMotion/CoreMotion.h>
 
 #ifdef RCT_NEW_ARCH_ENABLED
-#import "RNYuHealthSpec.h"
+#import <RNYuHealthSpec/RNYuHealthSpec.h>
 
 #if __has_include(<React/RCTEventEmitter.h>)
 #import <React/RCTEventEmitter.h>
diff --git a/node_modules/@yu-life/react-native-yu-health/ios/YuHealth.mm b/node_modules/@yu-life/react-native-yu-health/ios/YuHealth.mm
index dfaea2b..20a43d2 100644
--- a/node_modules/@yu-life/react-native-yu-health/ios/YuHealth.mm
+++ b/node_modules/@yu-life/react-native-yu-health/ios/YuHealth.mm
@@ -3,7 +3,7 @@
 #ifdef RCT_NEW_ARCH_ENABLED
 #import <ReactCommon/RCTTurboModule.h>
 #import <React/RCTUtils.h>
-#import "RNYuHealthSpec.h"
+#import <RNYuHealthSpec/RNYuHealthSpec.h>
 #endif
 
 #import "YuHealth.h"
@@ -147,60 +147,60 @@ - (void)requestPermissions:(nonnull NSArray *)capabilities provider:(nonnull NSS
 }
 
 - (void)activityQuery:(JS::NativeYuHealth::IActivityQueryNativeParams &)params resolve:(nonnull RCTPromiseResolveBlock)resolve reject:(nonnull RCTPromiseRejectBlock)reject {
+    NSDictionary *options = @{
+        @"startTime": params.startTime(),
+        @"endTime": params.endTime()
+    };
     dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        NSDictionary *options = @{
-            @"startTime": params.startTime(),
-            @"endTime": params.endTime()
-        };
         [self activityQuery:options resolver:resolve rejector:reject];
     });
 }
 
 
 - (void)aggregateQuery:(JS::NativeYuHealth::IAggregateQueryNativeParams &)params resolve:(nonnull RCTPromiseResolveBlock)resolve reject:(nonnull RCTPromiseRejectBlock)reject { 
+    NSDictionary *options = @{
+        @"capability": params.dataType(),
+        @"startTime": params.startTime(),
+        @"endTime": params.endTime()
+    };
     dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        NSDictionary *options = @{
-            @"capability": params.dataType(),
-            @"startTime": params.startTime(),
-            @"endTime": params.endTime()
-        };
         [self aggregateQuery:options resolver:resolve rejector:reject];
     });
 }
 
 - (void)queryPedometerFromDate:(JS::NativeYuHealth::IQueryPedometerFromDateNativeParams &)params resolve:(nonnull RCTPromiseResolveBlock)resolve reject:(nonnull RCTPromiseRejectBlock)reject { 
+    NSDictionary *options = @{
+        @"startTime": params.startTime(),
+        @"endTime": params.endTime()
+    };
     dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        NSDictionary *options = @{
-            @"startTime": params.startTime(),
-            @"endTime": params.endTime()
-        };
         [self queryPedometerFromDate:options resolver:resolve rejector:reject];
     });
 }
 
 - (void)sampleQuery:(JS::NativeYuHealth::ISampleQueryNativeParams &)params resolve:(nonnull RCTPromiseResolveBlock)resolve reject:(nonnull RCTPromiseRejectBlock)reject { 
+    NSDictionary *options = @{
+        @"capability": params.dataType(),
+        @"startTime": params.startTime(),
+        @"endTime": params.endTime()
+    };
     dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        NSDictionary *options = @{
-            @"capability": params.dataType(),
-            @"startTime": params.startTime(),
-            @"endTime": params.endTime()
-        };
         [self sampleQuery:options resolver:resolve rejector:reject];
     });
 }
 
 - (void)startPedometer:(JS::NativeYuHealth::IStartPedometerNativeParams &)params resolve:(nonnull RCTPromiseResolveBlock)resolve reject:(nonnull RCTPromiseRejectBlock)reject {
+    NSDictionary *options = @{
+        @"startTime": params.startTime(),
+        @"endTime": params.endTime()
+    };
     dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        NSDictionary *options = @{
-            @"startTime": params.startTime(),
-            @"endTime": params.endTime()
-        };
         [self startPedometer:options resolver:resolve rejector:reject];
         resolve(@(YES));
     });
 }
 
-- (void)startForegroundService:(JS::NativeYuHealth::IForegroundNotificationConfig &)notificationConfig resolve:(nonnull RCTPromiseResolveBlock)resolve reject:(nonnull RCTPromiseRejectBlock)reject {
+- (void)startForegroundService:(JS::NativeYuHealth::IForegroundServiceConfig &)serviceConfig resolve:(nonnull RCTPromiseResolveBlock)resolve reject:(nonnull RCTPromiseRejectBlock)reject {
     resolve(@(YES));
 }
 
