diff --git a/src/artifacts/templates/artifact/FileArtifact.js b/src/artifacts/templates/artifact/FileArtifact.js
index 65d23d4a8a62f0f7cccaadcb6d3d72d019cfd7b7..cece8e6c21e7185f63f1e72ee2d0d054356ac1ce 100644
--- a/src/artifacts/templates/artifact/FileArtifact.js
+++ b/src/artifacts/templates/artifact/FileArtifact.js
@@ -88,6 +88,12 @@ class FileArtifact extends Artifact {
       return true;
     }
 
+    if (await fs.exists(destination)) {
+      logger.debug({ event: 'MOVE_FILE_OVERWRITE' }, `moving "${source}" to ${destination}`);
+      await fs.move(source, destination, { overwrite: true });
+      return true;
+    }
+
     logger.warn({ event: 'MOVE_FILE_EXISTS' }, `cannot overwrite: "${source}" => "${destination}"`);
     await fs.remove(source);
     return false;
