diff --git a/node_modules/expo-image/ios/ExpoImage.podspec b/node_modules/expo-image/ios/ExpoImage.podspec
index de8bc65..9b63775 100644
--- a/node_modules/expo-image/ios/ExpoImage.podspec
+++ b/node_modules/expo-image/ios/ExpoImage.podspec
@@ -16,9 +16,9 @@ Pod::Spec.new do |s|
   s.static_framework = true
 
   s.dependency 'ExpoModulesCore'
-  s.dependency 'SDWebImage', '~> 5.19.1'
-  s.dependency 'SDWebImageWebPCoder', '~> 0.14.6'
-  s.dependency 'SDWebImageAVIFCoder', '~> 0.11.0'
+  s.dependency 'SDWebImage', '~> 5.11.1'
+  s.dependency 'SDWebImageWebPCoder', '~> 0.8.4'
+  s.dependency 'SDWebImageAVIFCoder', '~> 0.10.1'
   s.dependency 'SDWebImageSVGCoder', '~> 1.7.0'
   s.dependency 'libavif/libdav1d'
 
diff --git a/node_modules/expo-image/ios/ImageModule.swift b/node_modules/expo-image/ios/ImageModule.swift
index 330f460..55054b3 100644
--- a/node_modules/expo-image/ios/ImageModule.swift
+++ b/node_modules/expo-image/ios/ImageModule.swift
@@ -129,7 +129,7 @@ public final class ImageModule: Module {
       }
 
       urls.forEach { url in
-        SDWebImagePrefetcher.shared.prefetchURLs([url], context: context, progress: nil, completed: { _, skipped in
+        SDWebImagePrefetcher.shared.prefetchURLs([url], progress: nil, completed: { _, skipped in
           if skipped > 0 && !failed {
             failed = true
             promise.resolve(false)
diff --git a/node_modules/expo-image/ios/ImageView.swift b/node_modules/expo-image/ios/ImageView.swift
index 2d331d8..57c9ad5 100644
--- a/node_modules/expo-image/ios/ImageView.swift
+++ b/node_modules/expo-image/ios/ImageView.swift
@@ -145,15 +145,15 @@ public final class ImageView: ExpoView {
 
     // It seems that `UIImageView` can't tint some vector graphics. If the `tintColor` prop is specified,
     // we tell the SVG coder to decode to a bitmap instead. This will become useless when we switch to SVGNative coder.
-    if imageTintColor != nil {
-      context[.imageDecodeOptions] = [
-        SDImageCoderOption.webImageContext: [
-          "svgPrefersBitmap": true,
-          "svgImageSize": sdImageView.bounds.size,
-          "svgImagePreserveAspectRatio": true
-        ]
-      ]
-    }
+    // if imageTintColor != nil {
+    //   context[.imageDecodeOptions] = [
+    //     SDImageCoderOption.webImageContext: [
+    //       "svgPrefersBitmap": true,
+    //       "svgImageSize": sdImageView.bounds.size,
+    //       "svgImagePreserveAspectRatio": true
+    //     ]
+    //   ]
+    // }
 
     if source.isCachingAllowed {
       let sdCacheType = cachePolicy.toSdCacheType().rawValue
