diff --git a/node_modules/expo-image/ios/ExpoImage.podspec b/node_modules/expo-image/ios/ExpoImage.podspec
index d9ee9d3..043f48a 100644
--- a/node_modules/expo-image/ios/ExpoImage.podspec
+++ b/node_modules/expo-image/ios/ExpoImage.podspec
@@ -16,8 +16,8 @@ Pod::Spec.new do |s|
   s.static_framework = true
 
   s.dependency 'ExpoModulesCore'
-  s.dependency 'SDWebImage', '~> 5.17.0'
-  s.dependency 'SDWebImageWebPCoder', '~> 0.13.0'
+  s.dependency 'SDWebImage', '~> 5.11.1'
+  s.dependency 'SDWebImageWebPCoder', '~> 0.8.4'
   s.dependency 'SDWebImageAVIFCoder', '~> 0.10.1'
   s.dependency 'SDWebImageSVGCoder', '~> 1.7.0'
 
diff --git a/node_modules/expo-image/ios/ImageModule.swift b/node_modules/expo-image/ios/ImageModule.swift
index c914ad8..8780c35 100644
--- a/node_modules/expo-image/ios/ImageModule.swift
+++ b/node_modules/expo-image/ios/ImageModule.swift
@@ -116,7 +116,7 @@ public final class ImageModule: Module {
       var failed = false
 
       urls.forEach { url in
-        SDWebImagePrefetcher.shared.prefetchURLs([url], context: context, progress: nil, completed: { loaded, skipped in
+        SDWebImagePrefetcher.shared.prefetchURLs([url], progress: nil, completed: { loaded, skipped in
           if skipped > 0 && !failed {
             failed = true
             promise.resolve(false)
diff --git a/node_modules/expo-image/ios/ImageView.swift b/node_modules/expo-image/ios/ImageView.swift
index 245d5c5..64ef98e 100644
--- a/node_modules/expo-image/ios/ImageView.swift
+++ b/node_modules/expo-image/ios/ImageView.swift
@@ -136,15 +136,15 @@ public final class ImageView: ExpoView {
 
     // It seems that `UIImageView` can't tint some vector graphics. If the `tintColor` prop is specified,
     // we tell the SVG coder to decode to a bitmap instead. This will become useless when we switch to SVGNative coder.
-    if imageTintColor != nil {
-      context[.imageDecodeOptions] = [
-        SDImageCoderOption.webImageContext: [
-          "svgPrefersBitmap": true,
-          "svgImageSize": sdImageView.bounds.size,
-          "svgImagePreserveAspectRatio": true
-        ]
-      ]
-    }
+//    if imageTintColor != nil {
+//      context[.imageDecodeOptions] = [
+//        SDImageCoderOption.webImageContext: [
+//          "svgPrefersBitmap": true,
+//          "svgImageSize": sdImageView.bounds.size,
+//          "svgImagePreserveAspectRatio": true
+//        ]
+//      ]
+//    }
 
     if source.isCachingAllowed {
       let sdCacheType = cachePolicy.toSdCacheType().rawValue
