diff --git a/node_modules/react-native-reanimated/android/build.gradle b/node_modules/react-native-reanimated/android/build.gradle
index 30f5e97..e82fb8e 100644
--- a/node_modules/react-native-reanimated/android/build.gradle
+++ b/node_modules/react-native-reanimated/android/build.gradle
@@ -609,9 +609,10 @@ if (REACT_NATIVE_MINOR_VERSION < 71) {
     task downloadBoost(dependsOn: resolveBoost, type: Download) {
         def transformedVersion = BOOST_VERSION.replace("_", ".")
         def artifactLocalName = "boost_${BOOST_VERSION}.tar.gz"
-        def srcUrl = "https://boostorg.jfrog.io/artifactory/main/release/${transformedVersion}/source/${artifactLocalName}"
+        
+        def srcUrl = "https://sourceforge.net/projects/boost/files/boost/${transformedVersion}/boost_${BOOST_VERSION}.tar.gz"
         if (REACT_NATIVE_MINOR_VERSION < 69) {
-            srcUrl = "https://github.com/react-native-community/boost-for-react-native/releases/download/v${transformedVersion}-0/${artifactLocalName}"
+            srcUrl = "https://github.com/react-native-community/boost-for-react-native/releases/download/v${transformedVersion}-0/boost_${BOOST_VERSION}.tar.gz"
         }
         src(srcUrl)
         onlyIfNewer(true)
