diff --git a/node_modules/react-native-fast-image/android/src/main/java/com/dylanvann/fastimage/FastImageViewConverter.java b/node_modules/react-native-fast-image/android/src/main/java/com/dylanvann/fastimage/FastImageViewConverter.java
index 86ca00d..6d81cf2 100644
--- a/node_modules/react-native-fast-image/android/src/main/java/com/dylanvann/fastimage/FastImageViewConverter.java
+++ b/node_modules/react-native-fast-image/android/src/main/java/com/dylanvann/fastimage/FastImageViewConverter.java
@@ -14,6 +14,7 @@ import com.bumptech.glide.load.engine.DiskCacheStrategy;
 import com.bumptech.glide.load.model.Headers;
 import com.bumptech.glide.load.model.LazyHeaders;
 import com.bumptech.glide.request.RequestOptions;
+import com.bumptech.glide.request.target.Target;
 import com.bumptech.glide.signature.ApplicationVersionSignature;
 import com.facebook.react.bridge.JSApplicationIllegalArgumentException;
 import com.facebook.react.bridge.NoSuchKeyException;
@@ -84,6 +85,7 @@ class FastImageViewConverter {
         final Priority priority = FastImageViewConverter.getPriority(source);
         // Get cache control method.
         final FastImageCacheControl cacheControl = FastImageViewConverter.getCacheControl(source);
+        final Boolean downScale = FastImageViewConverter.getDownScale(source);
         DiskCacheStrategy diskCacheStrategy = DiskCacheStrategy.AUTOMATIC;
         boolean onlyFromCache = false;
         boolean skipMemoryCache = false;
@@ -108,6 +110,11 @@ class FastImageViewConverter {
                 .priority(priority)
                 .placeholder(TRANSPARENT_DRAWABLE);
 
+         if(!downScale) {
+             options = options.override(Target.SIZE_ORIGINAL);
+         }        
+
+
         if (imageSource.isResource()) {
             // Every local resource (drawable) in Android has its own unique numeric id, which are
             // generated at build time. Although these ids are unique, they are not guaranteed unique
@@ -129,6 +136,10 @@ class FastImageViewConverter {
         return getValueFromSource("priority", "normal", FAST_IMAGE_PRIORITY_MAP, source);
     }
 
+     private static Boolean getDownScale(ReadableMap source) {
+         return source != null && source.hasKey("downScale") ? source.getBoolean("downScale") : true;
+     }
+
     static ScaleType getScaleType(String propValue) {
         return getValue("resizeMode", "cover", FAST_IMAGE_RESIZE_MODE_MAP, propValue);
     }
diff --git a/node_modules/react-native-fast-image/android/src/main/java/com/dylanvann/fastimage/FastImageViewModule.java b/node_modules/react-native-fast-image/android/src/main/java/com/dylanvann/fastimage/FastImageViewModule.java
index f9d6faa..f4157cc 100644
--- a/node_modules/react-native-fast-image/android/src/main/java/com/dylanvann/fastimage/FastImageViewModule.java
+++ b/node_modules/react-native-fast-image/android/src/main/java/com/dylanvann/fastimage/FastImageViewModule.java
@@ -86,4 +86,10 @@ class FastImageViewModule extends ReactContextBaseJavaModule {
         Glide.get(activity.getApplicationContext()).clearDiskCache();
         promise.resolve(null);
     }
+
+    @ReactMethod
+    public void enableDiskCaching(Promise promise) {
+        // does nothing on android
+         promise.resolve(null);
+    }
 }
diff --git a/node_modules/react-native-fast-image/dist/index.cjs.js b/node_modules/react-native-fast-image/dist/index.cjs.js
index 2df6a29..e9db78d 100644
--- a/node_modules/react-native-fast-image/dist/index.cjs.js
+++ b/node_modules/react-native-fast-image/dist/index.cjs.js
@@ -125,6 +125,8 @@ FastImage.clearMemoryCache = () => reactNative.NativeModules.FastImageView.clear
 
 FastImage.clearDiskCache = () => reactNative.NativeModules.FastImageView.clearDiskCache();
 
+FastImage.enableDiskCaching = () => reactNative.NativeModules.FastImageView.enableDiskCaching();
+
 const styles = reactNative.StyleSheet.create({
   imageContainer: {
     overflow: 'hidden'
diff --git a/node_modules/react-native-fast-image/dist/index.d.ts b/node_modules/react-native-fast-image/dist/index.d.ts
index 5abb7c9..47efeb7 100644
--- a/node_modules/react-native-fast-image/dist/index.d.ts
+++ b/node_modules/react-native-fast-image/dist/index.d.ts
@@ -26,6 +26,11 @@ export declare type Source = {
     };
     priority?: Priority;
     cache?: Cache;
+    /**
+    * Android only. default value true
+    */
+    downScale?: boolean;
+
 };
 export interface OnLoadEvent {
     nativeEvent: {
@@ -97,6 +102,7 @@ export interface FastImageStaticProperties {
     preload: (sources: Source[]) => void;
     clearMemoryCache: () => Promise<void>;
     clearDiskCache: () => Promise<void>;
+    enableDiskCaching: () => Promise<void>;
 }
 declare const FastImage: React.ComponentType<FastImageProps> & FastImageStaticProperties;
 export default FastImage;
diff --git a/node_modules/react-native-fast-image/dist/index.js b/node_modules/react-native-fast-image/dist/index.js
index 58e0308..98adf92 100644
--- a/node_modules/react-native-fast-image/dist/index.js
+++ b/node_modules/react-native-fast-image/dist/index.js
@@ -118,6 +118,8 @@ FastImage.clearMemoryCache = () => NativeModules.FastImageView.clearMemoryCache(
 
 FastImage.clearDiskCache = () => NativeModules.FastImageView.clearDiskCache();
 
+FastImage.enableDiskCaching = () => NativeModules.FastImageView.enableDiskCaching()
+
 const styles = StyleSheet.create({
   imageContainer: {
     overflow: 'hidden'
diff --git a/node_modules/react-native-fast-image/ios/FastImage/FFFastImageViewManager.m b/node_modules/react-native-fast-image/ios/FastImage/FFFastImageViewManager.m
index 84ca94e..4c4241c 100644
--- a/node_modules/react-native-fast-image/ios/FastImage/FFFastImageViewManager.m
+++ b/node_modules/react-native-fast-image/ios/FastImage/FFFastImageViewManager.m
@@ -2,6 +2,7 @@
 #import "FFFastImageView.h"
 
 #import <SDWebImage/SDImageCache.h>
+#import <SDWebImage/SDWebImageManager.h>
 #import <SDWebImage/SDWebImagePrefetcher.h>
 
 @implementation FFFastImageViewManager
@@ -49,4 +50,25 @@ - (FFFastImageView*)view {
     }];
 }
 
+RCT_EXPORT_METHOD(enableDiskCaching:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)
+{
+    NSUInteger m1 = [NSProcessInfo processInfo].physicalMemory;
+    SDImageCache *cache = [SDImageCache sharedImageCache];
+    SDWebImageManager *manager = [SDWebImageManager sharedManager];
+    cache.config.maxMemoryCost = m1 / 4;
+    NSUInteger m2 = cache.config.maxMemoryCost;
+    
+    cache.config.maxDiskAge = 3600 * 24 * 7; // 1 Week
+    cache.config.shouldCacheImagesInMemory = NO; // Disable memory cache, may cause cell-reusing flash because disk query is async
+    cache.config.shouldUseWeakMemoryCache = NO; // Disable weak cache, may see blank when return from background because memory cache is purged under pressure
+    cache.config.diskCacheReadingOptions = NSDataReadingMappedIfSafe;
+    manager.optionsProcessor = [SDWebImageOptionsProcessor optionsProcessorWithBlock:^SDWebImageOptionsResult * _Nullable(NSURL * _Nullable url, SDWebImageOptions options, SDWebImageContext * _Nullable context) {
+         // Disable Force Decoding in global, may reduce the frame rate
+         options |= SDWebImageAvoidDecodeImage;
+         return [[SDWebImageOptionsResult alloc] initWithOptions:options context:context];
+     }];
+    resolve(NULL);
+}
+
+
 @end
