diff --git a/node_modules/react-native/Libraries/AppDelegate/RCTAppSetupUtils.h b/node_modules/react-native/Libraries/AppDelegate/RCTAppSetupUtils.h
index 8c57761..ebc8556 100644
--- a/node_modules/react-native/Libraries/AppDelegate/RCTAppSetupUtils.h
+++ b/node_modules/react-native/Libraries/AppDelegate/RCTAppSetupUtils.h
@@ -12,6 +12,7 @@
 #ifdef __cplusplus
 
 #import <memory>
+#define USE_HERMES 1
 
 #if USE_HERMES
 #if __has_include(<jsireact/HermesExecutorFactory.h>)
