diff --git a/node_modules/react-native-video/android-exoplayer/src/main/java/com/brentvatne/receiver/AudioBecomingNoisyReceiver.java b/node_modules/react-native-video/android-exoplayer/src/main/java/com/brentvatne/receiver/AudioBecomingNoisyReceiver.java
index 82f8e28..42ea269 100644
--- a/node_modules/react-native-video/android-exoplayer/src/main/java/com/brentvatne/receiver/AudioBecomingNoisyReceiver.java
+++ b/node_modules/react-native-video/android-exoplayer/src/main/java/com/brentvatne/receiver/AudioBecomingNoisyReceiver.java
@@ -5,6 +5,7 @@ import android.content.Context;
 import android.content.Intent;
 import android.content.IntentFilter;
 import android.media.AudioManager;
+import android.os.Build;
 
 public class AudioBecomingNoisyReceiver extends BroadcastReceiver {
 
@@ -25,7 +26,11 @@ public class AudioBecomingNoisyReceiver extends BroadcastReceiver {
     public void setListener(BecomingNoisyListener listener) {
         this.listener = listener;
         IntentFilter intentFilter = new IntentFilter(AudioManager.ACTION_AUDIO_BECOMING_NOISY);
-        context.registerReceiver(this, intentFilter);
+        if (Build.VERSION.SDK_INT >= 34 && context.getApplicationInfo().targetSdkVersion >= 34) {
+            context.registerReceiver(this, intentFilter, Context.RECEIVER_EXPORTED);
+        } else {
+            context.registerReceiver(this, intentFilter);
+        }
     }
 
     public void removeListener() {
diff --git a/node_modules/react-native-video/ios/Video/RCTVideo.m b/node_modules/react-native-video/ios/Video/RCTVideo.m
index a757c08..4226db9 100644
--- a/node_modules/react-native-video/ios/Video/RCTVideo.m
+++ b/node_modules/react-native-video/ios/Video/RCTVideo.m
@@ -907,7 +907,7 @@ - (void)setRestoreUserInterfaceForPIPStopCompletionHandler:(BOOL)restore
 }
 
 - (void)setupPipController {
-  if (!_pipController && _playerLayer && [AVPictureInPictureController isPictureInPictureSupported]) {
+  if (!_pipController && _playerLayer && [AVPictureInPictureController isPictureInPictureSupported] && _pictureInPicture) {
     // Create new controller passing reference to the AVPlayerLayer
     _pipController = [[AVPictureInPictureController alloc] initWithPlayerLayer:_playerLayer];
     _pipController.delegate = self;
